src/coreclr/jit/gentree.cpp

```diff
@@ -7937,6 +7956,32 @@ GenTreeCall* Compiler::gtCloneExprCallHelper(GenTreeCall* tree,
     {
         // Create and initialize the fgArgInfo for our copy of the call tree
         copy->fgArgInfo = new (this, CMK_Unknown) fgArgInfo(copy, tree);
+
+        // set the cloned "return buffer" node, if exist.
+        if (tree->_retBufArg != nullptr)
+        {
+            unsigned index = tree->gtCallThisArg == nullptr ? 0 : 1;
+            while (index < tree->fgArgInfo->ArgCount())
+            {
+                fgArgTabEntry* entry = tree->fgArgInfo->GetArgEntry(index);
+                if (entry->nonStandardArgKind == NonStandardArgKind::FixedRetBuffer ||
+                    entry->nonStandardArgKind == NonStandardArgKind::None)
+                {
+                    // found ret buffer
+                    GenTree* argNodeCopy = copy->fgArgInfo->GetArgEntry(index)->GetNode();
+                    if (argNodeCopy->OperIs(GT_ADDR))
+                    {
+                        GenTree* addrNodeCopy = argNodeCopy->AsOp()->gtGetOp1();
+                        assert(addrNodeCopy->OperIs(GT_LCL_VAR));
+
+                        copy->_retBufArg = addrNodeCopy->AsLclVar();
+                    }
+                    break;
+                }
+                index++;
```

Comment:
This copying code does not handle the field case and also looks to not handle the late arg case.

It does not seem right that this copying code should care about this, let me think about this for a bit, a better solution should be possible.

Reply:
Right. I think I am jitstress failures around that. I was looking for some existing method that would help me find the cloned node from the original node, but seems there is no such thing currently.

Reply:
Don't think this is needed anymore given that I am storing `Use*`.